warp-core/src/main/scala/com/workday/warp/junit/HasTestId.scala

```diff
@@ -16,9 +16,9 @@ import scala.util.Try
   */
 trait HasTestId {
```

Comment:
this will take a more primary role in our 5.x release.

my plan is to change many of our `testId: String` usages to usages of this trait with accompanying implicits for converting various things (TestInfo, ExecutionContext, etc) into instances of `HasTestId`

this will provide better support for annotation lookup since we'll have a reference to the test class and methods in scope